### PR TITLE
Added missing scenarios for snippet feature:

### PR DESCRIPTION
**Pull request by landrysoules**
- update
- delete
- create
- show

### CODE REVIEW

File: features/step_definitions/snippet_steps.rb [hunk 1]
```diff
@@ -15,10 +16,68 @@
...
-Then(/^I should see a success message$/) do
-  expect(page).to have_content("Snipet was successfully created.")
+Then(/^I should see a "(.*?)" message$/) do |message_type|
+  if message_type.equal? "success"
+    expect(page).to have_content("Snipet was successfully created.")
+  elsif message_type.equal? "update success"
+    expect(page).to have_content("Snipet was successfully updated.")
+  elsif message_type.equal? "update delete"
+    expect(page).to have_content("Snipet was successfully destroyed.")
+  end
```
Comment (shiroyasha) [inline features/step_definitions/snippet_steps.rb]: you can use `case` here:

``` ruby
case message_type
when "success" 
  expect(page).to have_content("Snipet was successfully created.")
when "update success"
  expect(page).to have_content("Snipet was successfully updated.")
when "update delete"
  expect(page).to have_content("Snipet was successfully destroyed.")
end
```

File: features/step_definitions/snippet_steps.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 Given(/^I am authenticated$/) do
-  user = User.new email: "landry.soules@gmail.com", password: "1Password", username: "landry"
+  # FIXME: is it possible to use factorygirl here (build(:user)) in place of User.new ?
+  user = User.new email: "landry.soules@gmail.com", password: "7ninahartley7", username: "landry"
```
Comment (shiroyasha) [inline features/step_definitions/snippet_steps.rb]: Yes, it is even prefered :)

I think you can write `FactoryGirl.create(:user)`

File: features/step_definitions/snippet_steps.rb [hunk 3]
```diff
@@ -15,10 +16,68 @@
...
 end
+
+When(/^I go to my snippets page$/) do
+  visit new_snipet_path
+  fill_in "snipet_title", :with => "my first snippet"
+  fill_in "snipet_content", :with => "ls"
+  click_button "submit_snipet"
+
+  #FIXME: How can I get current_user id, in place of writing id ?
+  visit my_snipets_path(3)
```
Comment (shiroyasha) [inline features/step_definitions/snippet_steps.rb]: you can save the user in the "Given(/^I am authenticated$/)" step in an instance variable

``` ruby
Given(/^I am authenticated$/) do
  @user = ...
end
```

and then use it in this step with:

``` ruby
visit my_snipets_path(@user.id)
```

The trick here is that all the steps share a common object when your test executes

**Comment by shiroyasha** [comment]
Looks really good :+1:

I would probably write 3 steps instead of one `Then(/^I should see a "(.*?)" message$/)`, but I think this is also fine.

**Comment by rastasheep** [comment]
Yap as @shiroyasha said in case when you would have 3 separate steps, you would be able to update your features to include one step instead 2:

```
..
    Then I should see a "update delete" message
    And I should see snippets page
```

would become:

```
...
    Then I should see that i have updated snippet
```

(or something similar)

**Comment by shiroyasha** [comment]
It looks great, feel free to merge :+1:

**Comment by landrysoules** [comment]
Thanks ! :-)

**Pull request merged**